lib/api/api-request.js

```diff
@@ -30,6 +36,84 @@ class RequestResponse extends Readable {
   }
 }
 
+class RequestResponse {
+  constructor (resume, abort) {
+    this[kAbort] = abort
+    this[kResume] = resume
+    this[kBody] = null
+    this[kError] = undefined
+  }
+
+  [kPush] (chunk) {
+    if (!this[kBody]) {
+      this[kBody] = this.readableNodeStream()
+    }
+    this[kBody].push(chunk)
+  }
+
+  [kDestroy] (err) {
+    if (this[kBody]) {
+      util.destroy(this[kBody], err)
+    } else {
+      this[kError] = err
+    }
+  }
+
+  readableNodeStream () {
+    if (this[kBody]) {
+      throw new Error('consumed') // TODO: Proper error.
+    }
+    this[kBody] = new RequestNodeReadable(this[kResume], this[kAbort])
+    if (this[kError] !== undefined) {
+      util.destroy(this[kBody], this[kError])
+    }
+    return this[kBody]
+  }
+
+  readableWebStream () {
+    // TODO: Optimize.
+    return this.readableNodeStream().toWeb()
+  }
+
+  async blob () {
+    // TODO: Optimize.
+    const sources = []
+    for await (const chunk of this) {
+      // TOOD: max size?
+      sources.push(chunk)
+    }
+    return new Blob(sources)
+  }
+
+  async arrayBuffer () {
+    // TODO: Optimize.
+    const blob = await this.blob()
+    return await blob.arrayBuffer()
+  }
+
+  * [Symbol.asyncIterator] () {
+    // TODO: Optimize.
+    yield * this.readableNodeStream()
+  }
+
+  async text () {
+    // TODO: Optimize.
+    // TODO: Validate content-type req & res headers?
+    let ret = ''
+    for await (const chunk of this) {
+      // TOOD: max size?
+      ret += chunk
+    }
+    return JSON.parse(ret)
+  }
+
+  async json () {
+    // TODO: Optimize.
+    // TODO: Validate content-type req & res headers?
+    return JSON.parse(await this.text())
+  }
+}
+
```

Comment:
The idea here is that whatever way the user decides to consume the body it will always have the best possible performance, i.e. it shouldn't always go through `readableNodeStream()`. Instead each should have an adapter against the native API. 

I used `readableNodeStream` to start with to get the general functionality.